lib/ui/views/login_view.dart

```diff
@@ -62,14 +102,30 @@ class _Container extends ProviderWidget<LoginViewModel> {
 }
 
 class _EmailTextField extends ProviderWidget<LoginViewModel> {
+  final GlobalKey<FormState> formKey;
+  final TextEditingController controller;
+  final FocusNode currentFocusNode;
+  final FocusNode nextFocusNode;
+
+  _EmailTextField({
+    this.formKey,
+    @required this.controller,
+    this.currentFocusNode,
+    this.nextFocusNode,
+  });
+
   @override
   Widget build(BuildContext context, LoginViewModel model) {
     final local = AppLocalizations.of(context);
 
     return TextFormField(
-      controller: model.emailController,
+      controller: controller,
       validator: model.validateEmail,
-      onFieldSubmitted: (_) => model.passwordFocusNode.requestFocus(),
+      onFieldSubmitted: (_) => nextFocusNode.requestFocus(),
+      onChanged: (email) {
+        model.setEmail(email);
+        formKey.currentState?.validate();
```

Comment:
Either add the first ? back or make the formKey required

Reply:
formKey removed